crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -1207,6 +1220,18 @@ where
         &mut self,
         revert_until: BlockNumber,
     ) -> Result<Option<Chain>, CanonicalError> {
+        if self
+            .externals
+            .provider_factory
+            .static_file_provider()
+            .get_highest_static_file_block(StaticFileSegment::Headers)
+            .unwrap_or_default() >
+            revert_until
+        {
+            // should only happen when an optimistic sync target was re-orged.
```

Comment:
can we add to the comment saying why only optimistic canonical reverts go past the highest static file header?